File: algorithm/src/main/java/baekjoon/p05532/Main.java
package baekjoon.p05532;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] inputs = new int[5];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        System.out.println(solve(inputs));
    }

    private static int solve(int[] inputs) {
        int l = inputs[0];
        int a = (inputs[1] + inputs[3] - 1) / inputs[3];
        int b = (inputs[2] + inputs[4] - 1) / inputs[4];

        return l - Math.max(a, b);
    }

}

File: algorithm/src/main/java/baekjoon/p05532/Main.kt
package baekjoon.p05532

import kotlin.math.ceil

fun main(args: Array<String>) {
    val inputs = Array(5) { readLine()!!.toInt() }
    println(solve(inputs))
}

private fun solve(inputs: Array<Int>): Int {
    val l = inputs[0]
    val a = ceil(inputs[1].toDouble() / inputs[3]).toInt()
    val b = ceil(inputs[2].toDouble() / inputs[4]).toInt()

    return l - maxOf(a, b)
}